src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -371,6 +385,9 @@ private ImmutableArray<TypeParameterSymbol> MakeTypeParameters(DiagnosticBag dia
                         ImmutableArray.Create(location),
                         ImmutableArray.Create(parameter.GetReference()));
 
+                // Force attribute binding for diagnostics
+                typeParameter.GetAttributesBag(diagnostics);
+
```

Comment:
You have in fact pre-empted my change to fix the circularity in binding the default value of parameter symbols. :)

My second PR relies on this one and also moves most of these checks to the GrabDiagnostics method, which I delay calling until the binder has finished creating the symbol.

I will attempt to push some of these changes back into this PR.